CorsixTH/Lua/app.lua

```diff
@@ -1553,10 +1554,32 @@ function App:checkCompatibility(save_version, demo_flag)
   else -- savegame newer than application
     err = _S.errors.compatibility_error.new_in_old
   end
+
   UILoadGame:loadError(err)
   return false
 end
 
+--! Compatibility function to work out the game's graphics set
+--!param state The savegame data
+function App:configureGraphicsSetFlag(state)
+  -- First eliminate all but the first campaign level (same in demo and full)
+  local self = state
```

Comment:
I don't get this line. The function parameters are both `self` and `state`, and here you make the original `self` disappear and rename `state` to a new `self`?

Huh?

Reply:
You might be looking at an old commit. The line should be now
`function App.configureGraphicsSetFlag(state)`

Reply:
Though arguably, there is no speific need to have `local self = state`, but I thought it might be less confusing to read. Potentially wrong.

Reply:
I agree with @Alberth289346, this is very strange. If it was to be self then just make it App:configureGraphicsSetFlags however reading the use, the object passed in is not self, and you shouldn't pretend it is.